docker/test_env/wait_strategies.go

```diff
@@ -0,0 +1,108 @@
+package test_env
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"time"
+
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+)
+
+type LogRegexStrategy struct {
+	timeout      *time.Duration
+	Pattern      *regexp.Regexp
+	Occurrence   int
+	PollInterval time.Duration
+}
+
+func NewLogRegexStrategy(pattern *regexp.Regexp) *LogRegexStrategy {
+	return &LogRegexStrategy{
+		Pattern:      pattern,
+		Occurrence:   1,
+		PollInterval: defaultPollInterval(),
+	}
+}
+
+func (ws *LogRegexStrategy) WithStartupTimeout(timeout time.Duration) *LogRegexStrategy {
+	ws.timeout = &timeout
+	return ws
+}
+
+// WithPollInterval can be used to override the default polling interval of 100 milliseconds
+func (ws *LogRegexStrategy) WithPollInterval(pollInterval time.Duration) *LogRegexStrategy {
+	ws.PollInterval = pollInterval
+	return ws
+}
+
+func (ws *LogRegexStrategy) WithOccurrence(o int) *LogRegexStrategy {
+	// the number of occurrence needs to be positive
+	if o <= 0 {
+		o = 1
+	}
+	ws.Occurrence = o
+	return ws
+}
+
+// WaitUntilReady implements Strategy.WaitUntilReady
+func (ws *LogRegexStrategy) WaitUntilReady(ctx context.Context, target tcwait.StrategyTarget) (err error) {
```

Comment:
Can we add a comment about what we are searching for and why it's so specific?

Reply:
I think I will remove it, because I found a better way to make sure that an epoch is finalized.